jujuclient/file.go

```diff
@@ -266,10 +285,10 @@ func (s *store) SetCurrentController(name string) error {
 	if _, ok := controllers.Controllers[name]; !ok {
 		return errors.NotFoundf("controller %v", name)
 	}
-	if controllers.CurrentController == name {
-		return nil
+	if controllers.HasControllerChangedOnPreviousSwitch = controllers.CurrentController != name; controllers.HasControllerChangedOnPreviousSwitch {
```

Comment:
I think this is trying to be too clever. I would rather you explicitly set HasControllerChangedOnPreviousSwitch